plugins/inventory/lxd.py

```diff
@@ -0,0 +1,1066 @@
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: lxd
+    short_description: Returns Ansible inventory from lxd host
+    description:
+        - Get inventory from the lxd.
+        - Uses a YAML configuration file that ends with 'lxd.(yml|yaml)'.
+    version_added: "2.5.0"
+    author: "Frank Dornheim (@conloos)"
+    options:
+        plugin:
+            description: Token that ensures this is a source file for the 'lxd' plugin.
+            required: true
+            choices: [ 'community.general.lxd' ]
+        url:
+            description:
+            - The unix domain socket path or the https URL for the lxd server.
+            - Sockets in filesystem have to start with C('unix:').
+            - Mostly C(unix:/var/lib/lxd/unix.socket) or C(unix:/var/snap/lxd/common/lxd/unix.socket).
+            default: unix:/var/snap/lxd/common/lxd/unix.socket
+            required: false
+            type: str
+        client_key:
+            description:
+            - The client certificate key file path.
+            required: false
+            aliases: [ key_file ]
+            default: $HOME/.config/lxc/client.key
+            type: path
+        client_cert:
+            description:
+            - The client certificate file path.
+            required: false
+            aliases: [ cert_file ]
+            default: $HOME/.config/lxc/client.crt
+            type: path
+        trust_password:
+            description:
+            - The client trusted password.
+            - You need to set this password on the lxd server before
+                running this module using the following command
+                C('lxc config set core.trust_password <some random password>')
+                See U(https://www.stgraber.org/2016/04/18/lxd-api-direct-interaction/).
+            - If I(trust_password) is set, this module send a request for authentication before sending any requests.
+            required: false
+            type: str
+        state:
+            description: Filter the container according to the current status.
+            required: false
+            type: str
+            default: none
+            choices: [ 'STOPPED', 'STARTING', 'RUNNING', 'none' ]
+        prefered_container_network_interface:
+            description:
+            - If a container has multiple network interfaces, select which one is the prefered as pattern.
+            - Combined with the first number that can be found e.g. 'eth' + 0.
+            required: false
+            type: str
+            default: eth
+        prefered_container_network_family:
+            description:
+            - If a container has multiple network interfaces, which one is the prefered by family.
+            - Specify C(inet) for IPv4 and C(inet6) for IPv6.
+            required: false
+            type: str
+            default: inet
+            choices: [ 'inet', 'inet6' ]
+        groups:
```

Comment:
Is this option really necessary? First it prevents that the `Constructed` interface is ever implemented, and also with the `Constructed` interface (https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/doc_fragments/constructed.py), creating groups from properties is pretty simple.

Reply:
Hi Felix,

i don't understand your hint.
I put a lot of work in the dynamic grouping of hosts e.g. it is possible to create groups by network ranges like 192.168.1.0/24.
So can you explaine how to implement that features without the "groups" option?
As i can see it in the other inventorys it seams that the creating of groups is more static.


Thanks Frank

Reply:
Hmm, I think network ranges are the only part that cannot be done so easily. You might be able to achieve something similar with the `ipaddr` filters (https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters_ipaddr.html), but you probably cannot do everything. (Here are examples of what can be done: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/constructed_inventory.html#examples)

In any case, you should rename this option so you do not make it impossible to later also implement `Constructed` for this plugin. If you use the same name as one of its parameters, you prevent implementing parts of it.

How about naming this `value_groups`, since these groups are created from specific values?